src/javascript/campaign-url-builder/components/campaign-url.js

```diff
@@ -33,16 +36,50 @@ const INSTRUCTIONS_TEXT =
     'automatically generated for you here.';
 
 
+const gaSendEvent = ({category, action, label}) =>
+  gaAll('send', 'event', {
+    eventCategory: category,
+    eventAction: action,
+    eventLabel: label,
+  });
+
+
 /**
  * A component that renders the generated campaign URL.
  */
 export default class CampaignUrl extends React.Component {
+  /**
+   * Constructor for a campain URL component.
+   *
+   * @param {Object} props the initial props sent to the Rendered
+   * Campaign URL
+   */
+  constructor(props) {
+    super(props);
+
+    const {element, eventLabel} = renderProblematic(props.url);
+    if (!isNil(element)) {
+      gaSendEvent({
+        category: 'Campaign URL',
+        action: 'Problematic URL',
+        label: eventLabel,
+      });
+    }
 
-  state = {
```

Comment:
Because I added a constructor, and part of what the constructor does is call `renderProblematic` with the props and populate the state with the results of that function call. If I wanted to be *really* more correct about it, I'd use a reselect selector instead, single the element and eventLabel are "derrived data" and not authoritative.

Reply:
Gotcha.